apps/aehyperchains/src/aehc_connector.erl

```diff
@@ -1,36 +1,98 @@
-                       tx_hash().
-broadcast(Con, From, TxPayload, To) ->
-    %% Additional params (like value, fee) are parent related and can be passed via appropriate config;
-    Res = Con:broadcast(From, TxPayload, To), true = is_binary(Res),
-    Res.
+-spec connector() -> connector().
+connector() ->
+    ok.
 
--spec verify(Con::connector(), TxHash::tx_hash()) -> 
-                    boolean().
-verify(Con, TxHash) ->
-    Res = Con:verify(TxHash), true = is_boolean(Res), 
-    Res.
+%%%===================================================================
+%%%  Parent chain simplified proto
+%%%===================================================================
 
--spec subscribe(Con::connector(), Module::atom()) -> ok.
-subscribe(Con, Module) ->
-    ok = Con:subscribe(Module).
+-record(tx, {
+             sender :: binary(),
+             recipient :: binary(),
+             payload :: binary()
+            }).
+
+-type tx() :: #tx{}.
+
+-record(header, {
+                 hash :: binary(),
+                 number = 0 :: integer()
+                }).
+
+-type header() :: #header{}.
+
+-record(block, {
+                header :: header(),
+                txs :: [tx()]
+               }).
+
+-type block() :: #block{}.
+
+-spec tx(Sender::binary(), Recipient::binary(), Payload::binary()) -> tx().
+tx(Sender, Recipient, Payload) when 
+      is_binary(Sender), is_binary(Recipient), is_binary(Payload) ->
+    #tx{ sender=Sender, recipient=Recipient, payload=Payload }.
+
+-spec header(Hash::binary(), Num::integer()) -> header().
+header(Hash, Num) when 
+      is_binary(Hash), is_integer(Num) ->
+    #header{ hash=Hash, number=Num }.
+
+-spec block(Header::header(), Txs::[tx()]) -> block().
+block(Header, Txs) when
+      is_record(Header, header), is_list(Txs) ->
+    #block{ header=Header, txs=Txs }.
+
+%%%===================================================================
+%%%  Parent chain interface
+%%%===================================================================
+
+-spec send_tx(Tx::aetx:tx()) ->
+                    {ok, TxHash::binary()} | {error, {term(), term()}}.
+send_tx(Tx) ->
+    Con = aehc_chain_sim_connector, %% TODO To ask via config;
+    try
+        Res = Con:send_tx(Tx), true = is_binary(Res),
+        {ok, Res}
+    catch E:R ->
+            {error, {E, R}}
+    end.
+
+-spec get_block(Num::integer()) ->
+                       {ok, block()} | {error, {term(), term()}}.
+get_block(Num) ->
+    Con = aehc_chain_sim_connector, %% TODO To ask via config;
+    try 
+        Res = Con:get_block(Num), true = is_record(Res, block),
+        {ok, Res}
+    catch E:R ->
+            {error, {E, R}}
+    end.
+
+%%%===================================================================
+%%%  Parent chain events
+%%%===================================================================
+
+-spec subscribe(aec_events:event()) -> true.
+subscribe(Event) ->
+    aec_events:subscribe(Event).
```

Comment:
You need to change typespecs in aec_events anyway. Why not use aec_events directly?

Reply:
Thanks for the review, I got you. The purpose behind to restrict knowledge of connector's developer about internal message bus. All stuff which a client  is needed is to call aehc_connector:pubslish/1 (without knowledge about what we use today aec_events or something other)